lake/root.go

```diff
@@ -189,6 +211,14 @@ func (r *Root) OpenPool(ctx context.Context, id ksuid.KSUID) (*Pool, error) {
 	return poolRef.Open(ctx, r.path)
 }
 
+func (r *Root) OpenPoolByID(ctx context.Context, id ksuid.KSUID) (*Pool, error) {
+	poolRef := r.LookupPool(ctx, id)
+	if poolRef == nil {
+		return nil, fmt.Errorf("%s: pool not found", id)
```

Comment:
Should return ErrPoolNotFound.

Reply:
I deleted this function so not relevant anymore.  But using OpenPool() and ErrPoolNotFound means all the callers have to prepend the pool name/ID.

Reply:
You can wrap it like `fmt.Errorf("%s: %w", ErrPoolNotFound)`, its important to do this so you can introspect the error. The server needs this to know if it needs to return 404.